### ISSUE TITLE
Manual signature necessary?

### ISSUE THREAD

**Issue by hugobowne** [datacamp/pythonwhat#108]
I need to use a manual signature is a case that @filipsch thinks I should not. The following SCT throws this error:

<img width="642" alt="screenshot 2016-10-04 10 26 42" src="https://cloud.githubusercontent.com/assets/4182382/19078291/a7860394-8a1d-11e6-9e09-ffe1258873fc.png">

*** =pre_exercise_code
```{python}
import pandas as pd

# Make DataFrame of iris data
import sklearn.datasets
data = sklearn.datasets.load_iris()
df = pd.DataFrame(data.data, columns=data.feature_names)
df['species'] = data.target
df = df.rename(columns={'petal length (cm)': 'petal_length'})
for i in [0, 1, 2]:
    df.loc[df['species']==i, 'species'] = data.target_names[i]

df = df[['species', 'petal_length']]

# Extract Series that has versicolor petal lengths
versicolor_petal_length = df.loc[df.species=='versicolor',
                                 'petal_length'].values
```

*** =solution
```{python}
# Import plotting modules
import matplotlib.pyplot as plt
import seaborn as sns

# Set default Seaborn style
sns.set()

# Plot histogram of versicolor petal lengths
_ = plt.hist(versicolor_petal_length)

# Show histogram
plt.show()

```

*** =sct
```{python}

test_import("matplotlib.pyplot")

test_import("seaborn")

test_function("seaborn.set")

#sig = sig_from_params(param("x", param.POSITIONAL_OR_KEYWORD))

test_function_v2("matplotlib.pyplot.hist", params=["x"], not_called_msg="Did you create the histogram using `plt.hist`?", incorrect_msg="Did you pass the correct argument to `plt.hist`?")


#test_object("_", undefined_msg="Did you create the histogram as `_`?", incorrect_msg="Did you pass the correct argument to 'plt.hist'?")

test_function("matplotlib.pyplot.show")

success_msg("Great work!")
```

**Comment by machow** [comment]
This is caused by test_function_v2 expecting `matplotlib` to have been imported at the module level. If I add `import matplotlib` to the submitted code, then it works okay.

The issue is that when [get_signature](https://github.com/datacamp/pythonwhat/blob/master/pythonwhat/tasks.py#L276-L277) receives a `signature=None`, it uses a complex assortment of try/except blocks to figure out what the signature should be, and I'm guessing that it is looking for `matplotlib.pyplot.hist` when the environment only has `plt.hist`. 

@filipsch we should walk through this function tomorrow.

**Comment by filipsch** [comment]
@machow happy to!

**Comment by machow** [comment]
Original example should work now (tested in teach editor). Fixed in 84f734208171574196b17abeac19735453ce6b91
